com.unity.render-pipelines.universal/Runtime/2D/Light2D.cs

```diff
@@ -232,7 +238,30 @@ public float intensity
         /// The lights current intensity
         /// </summary>
         public float volumeOpacity => m_LightVolumeOpacity;
-        public Sprite lightCookieSprite => m_LightCookieSprite;
+
+        public Sprite lightCookieSprite
+        {
+            get
+            {
+                if (lightType == LightType.Sprite)
+                    return m_SpriteLightCookie;
+                else if (lightType == LightType.Point)
+                    return m_PointLightCookie;
+                else
+                    return null;
+            }
+            set
+            {
+                if (lightType == LightType.Sprite)
+                    m_SpriteLightCookie = value;
+                else if (lightType == LightType.Point)
+                    m_PointLightCookie = value;
+                else
+                    Debug.LogWarning("Unable to assign light cookie to " + lightType.ToString() + " light");
+            }
```

Comment:
Can we have two separate properties for this? For compatibility we can keep `lightCookieSprite`, but it should be read-only and marked as obsolete.